daemon/src/db.rs

```diff
@@ -348,6 +365,444 @@ impl Connection {
 
         Ok(ids)
     }
+
+    pub async fn insert_closed_cfd(&self, cfd: close_cfds::Cfd) -> Result<()> {
+        let mut conn = self.inner.acquire().await?;
+
+        let close_cfds::Cfd {
+            id,
+            position,
+            initial_price,
+            taker_leverage,
+            n_contracts,
+            counterparty_network_identity,
+            role,
+            opening_fee,
+            initial_funding_rate,
+            initial_tx_fee_rate,
+            fee_account,
+            lock,
+            commit,
+            collaborative_settlement,
+            non_collaborative_settlement,
+            refund,
+            close_status,
+            ..
+        } = cfd;
+
+        let (lock_txid, lock_dlc_vout) = match lock {
+            Some(Lock { txid, dlc_vout }) => (txid, dlc_vout),
+            None => bail!("Closed CFD without lock"),
+        };
+
+        let kind = match close_status {
+            CloseStatus::NotClosed => bail!("Cannot insert open CFD in closed CFDs table"),
+            CloseStatus::Collaborative => ClosedCfdKind::Collaborative,
+            CloseStatus::NonCollaborative => ClosedCfdKind::NonCollaborative,
+            CloseStatus::Refund => ClosedCfdKind::Refund,
```

Comment:
If you split the table layout into `closed_cfds` and `transactions`, this insertion logic should become a lot simpler because you just need to insert all transactions and don't care as much about why it was closed.

That reasoning can be done upon fetching a closed CFD from this table based on which transaction we were able to load.